internal/contour/cluster_test.go

```diff
@@ -33,7 +35,7 @@ func TestClusterCacheRecomputeService(t *testing.T) {
 	}{
 		"add unnamed service": {
 			oldObj: nil,
-			newObj: service("default", "kuard",
+			newObj: service("default", "kuard", nil,
```

Comment:
please add a new helper, something like serviceWithProtocol that takes the parameter which has been nil`d out here.

Reply:
I removed the parameter from `service` and included an additional `serviceWithAnnotations` in order to avoid the creation of nil-able protocol struct.  let me know if even so you want me to change with your proposed semantics.